wdl-engine/src/eval/v1/workflow.rs

```diff
@@ -804,28 +807,71 @@ impl WorkflowEvaluator {
         });
 
         // Evaluate the root graph to completion
-        Self::evaluate_subgraph(
-            state.clone(),
-            Scopes::ROOT_INDEX,
-            subgraph,
-            max_concurrency,
-            Arc::new(id.to_string()),
-            progress,
-        )
-        .await?;
-
-        // Take the output scope and return it
-        let mut outputs: Outputs = state.scopes.write().await.take(Scopes::OUTPUT_INDEX).into();
-        if let Some(section) = definition.output() {
-            let indexes: HashMap<_, _> = section
-                .declarations()
-                .enumerate()
-                .map(|(i, d)| (d.name().hashable(), i))
-                .collect();
-            outputs.sort_by(move |a, b| indexes[a].cmp(&indexes[b]))
-        }
+        let evaluation_result: EvaluationResult<()> = {
+            Self::evaluate_subgraph(
+                state.clone(),
+                Scopes::ROOT_INDEX,
+                subgraph,
+                max_concurrency,
+                Arc::new(id.to_string()),
+                progress,
+            )
+            .await
+        };
 
-        Ok(outputs)
+        let cleanup_result = match self.backend.cleanup(&effective_output_dir) {
```

Comment:
I think a lot of the logging logic here can ultimately be removed and we would just call:

```rust
let evaluation_result = ...;

if let Some(cleanup) = self.backend.cleanup(&effective_output_dir) {
    cleanup.await;
}

let mut outputs = ...;
Ok(outputs)
```

The `cleanup` implementation could potentially log any problems, but I don't want `cleanup` to cause `sprocket run` to fail if it ran into a problem.